src/safeposix/syscalls/net_calls.rs

```diff
@@ -1358,39 +1490,149 @@ impl Cage {
                     );
                 }
             }
+        //Otherwise, the write gaurd does not hold a FileDescriptor
         } else {
             return syscall_error(Errno::EBADF, "sendto", "invalid file descriptor");
         }
     }
 
+    /// ### Description
+    ///
+    /// `send_syscall` sends a message on a socket
+    ///
+    ///  The send() call may be used only when the socket is in a
+    ///  connected state (so that the intended recipient is known).
+    ///
+    /// ### Arguments
+    ///
+    /// it accepts four parameters:
+    /// * `fd` - the file descriptor of the sending socket
+    /// * `buf` - the message is found in buf
+    /// * `buflen` - the len of the message found in buf
+    /// * `flags` - bitwise OR of zero or more flags. Refer to man page to find
+    ///   possible args
+    ///
+    /// ### Returns
+    ///
+    /// On success, the call returns the number of bytes sent. On error, a
+    /// negative error number is returned, with the errorno set to represent
+    /// the corresponding error
+    ///
+    /// ### Errors
+    ///
+    /// These are some standard errors generated by the socket layer.
+    ///    Additional errors may be generated and returned from the
+    ///    underlying protocol modules; see their respective manual pages.
+    ///
+    /// * EACCES - (For UNIX domain sockets, which are identified by pathname)
+    ///   Write permission is denied on the destination socket file, or search
+    ///   permission is denied for one of the directories the path prefix.  (See
+    ///   path_resolution(7).)  (For UDP sockets) An attempt was made to send to
+    ///   a network/broadcast address as though it was a unicast address.
+    ///
+    /// ** EAGAIN or EWOULDBLOCK - The socket is marked nonblocking and the
+    ///   requested operation would block.  POSIX.1-2001 allows either error to
+    ///   be returned for this case, and does not require these constants to
+    ///   have the same value, so a portable application should check for both
+    ///   possibilities.
+    ///
+    /// ** EAGAIN - (Internet domain datagram sockets) The socket referred to by
+    ///   sockfd had not previously been bound to an address and, upon
+    ///   attempting to bind it to an ephemeral port, it was determined that all
+    ///   port numbers in the ephemeral port range are currently in use.  See
+    ///   the discussion of /proc/sys/net/ipv4/ip_local_port_range in ip(7).
+    ///
+    /// * EALREADY - Another Fast Open is in progress.
+    ///
+    /// ** EBADF - sockfd is not a valid open file descriptor.
```

Comment:
Why the difference in * vs **?

Reply:
At the bottom the errors, I wrote " ** Indicates the error may be returned from RustPOSIX." Maybe moving it to the top would be clearer?

Reply:
If an error cannot be returned, why is it listed?   Isn't this comment a description of this interface?

Reply:
Realistically all of these can _possibly_ be returned when were in INET mode because it calls send/recv on a raw socket. So instead of removing these or doing weird things with the * notation I think just making it clear that thats the behavior would be best.